File: backend/internal/models/fiber.go
package models

import "time"

type Fiber struct {
	ID          int       `json:"id" gorm:"primary_key;AUTO_INCREMENT"`
	Uuid        string    `json:"uuid" gorm:"column:uuid;unique;not null;type:varchar(36)"`
	Name        string    `json:"name" gorm:"column:name;not null"`
	Status      string    `json:"status" gorm:"column:status"`
	StockSortId string    `json:"stock_sort_id" gorm:"column:stock_sort_id"`
	SaleId      string    `json:"sale_id" gorm:"column:sale_id"`
	Deleted     bool      `json:"deleted" gorm:"column:deleted"`
	CreatedAt   time.Time `json:"created_at" gorm:"column:created_at"`
	UpdatedAt   time.Time `json:"updated_at" gorm:"column:updated_at"`
}

func (*Fiber) TableName() string {
	return "fibers"
}

type FiberRequest struct {
	Name        string `json:"name" validate:"required"`
	Status      string `json:"status" validate:"required"`
	StockSortId string `json:"stock_sort_id"`
}

type FiberResponse struct {
	Uuid        string    `json:"uuid" gorm:"column:uuid"`
	Name        string    `json:"name" gorm:"column:name"`
	Status      string    `json:"status" gorm:"column:status"`
	StockSortId string    `json:"stock_sort_id" gorm:"column:stock_sort_id"`
	Deleted     bool      `json:"deleted" gorm:"column:deleted"`
	CreatedAt   time.Time `json:"created_at" gorm:"column:created_at"`
	SaleCode    *string   `json:"sale_code" gorm:"column:sale_code"`
	SaleId      string    `json:"sale_id" gorm:"column:sale_id"`
}

type FiberPaginationResponse struct {
	Size   int             `json:"size"`
	PageNo int             `json:"page_no"`
	Total  int             `json:"total"`
	Data   []FiberResponse `json:"data"`
}

type BulkFiberRequest struct {
	Data []FiberRequest `json:"data"`
}

type FiberFilter struct {
	Size   int    `form:"size"`
	PageNo int    `form:"page_no"`
	Name   string `form:"name"`
	Status string `form:"status"`
}

type FiberAllocationRequest struct {
	ItemId    string `json:"item_id"`
	FiberId   string `json:"fiber_id"`
	FiberName string `json:"fiber_name"`
}

type FiberUsedList struct {
	FiberId   string `json:"uuid"`
	FiberName string `json:"name"`
}

type FiberStatistics struct {
	Total       int64   `json:"total"`
	Free        int64   `json:"free"`
	Used        int64   `json:"used"`
	Utilization float64 `json:"utilization_percentage"`
}

type FiberStats struct {
	Total int64 `gorm:"column:total"`
	Free  int64 `gorm:"column:free"`
	Used  int64 `gorm:"column:used"`
}

File: frontend/src/types/fiber.ts
export const STATUS_OPTIONS_LIST = [
    { key: '', label: 'Semua Status' },
    { key: 'FREE', label: 'Tersedia' },
    { key: 'USED', label: 'Digunakan' },
];

export const STATUS_MAP = {
    '': 'Semua Status',
    FREE: 'Tersedia',
    USED: 'Digunakan'
} as const;

type FiberStatus = 'FREE' | 'USED';

export interface FiberRequest {
    name: string;
    status: FiberStatus;
}

export interface FiberResponse {
    uuid: string;
    name: string;
    status: FiberStatus;
    deleted: boolean;
    created_at: string;
}

export interface FiberPaginationResponse {
    size: number;
    page_no: number;
    total: number;
    data: FiberResponse[];
}

export interface BulkFiberRequest {
    data: FiberRequest[];
}

export interface FiberFilter {
    size?: number;
    page_no?: number;
    name?: string;
    status?: string;
}

export interface FiberList {
    uuid: string;
    name: string;
}